src/rgw/rgw_lc.cc

```diff
@@ -1073,22 +1073,17 @@ class LCOpAction_CurrentExpiration : public LCOpAction {
       return false;
     }
     if (o.is_delete_marker()) {
-      if (oc.next_key_name) {
```

Comment:
I think this change may be removing part of required lookahead logic, and it does it without comment (oc.next_key_name is a std::optional<std::string> iirc, propagated from the lc-obj-lister)--I'd be concerned that this part is not correct

Reply:
I noticed that `next_key_name` is checked inside `next_has_same_name()`, please see https://github.com/ceph/ceph/blob/v19.0.0/src/rgw/rgw_lc.cc#L502, so I removed this condition and left all to `next_has_same_name()`.